File: pc-square-io-separator-java/src/main/java/com/mj/pc/square/io/separator/java/util/TestCase.java
package com.mj.pc.square.io.separator.java.util;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * @author Montaser Sbaih
 * @version 1.0
 * @email montaser.jjs@gmail.com
 * @phone +962-786258874
 * @since 20-10-2022
 */

@Getter
@NoArgsConstructor
@AllArgsConstructor
public final class TestCase {

    private String input;

    private String output;

    @Override
    public String toString() {
        return String.format("%s\n-\n%s", input, output);
    }

    public static TestCase parse(String testCase) {
        String[] IO = testCase.split("\n*\\-+\n*");
        return new TestCase(IO[0], IO[1]);
    }
}

File: pc-square-io-separator-kotlin/src/main/kotlin/com/mj/pc/square/io/separator/kotlin/util/TestCase.kt
package com.mj.pc.square.io.separator.kotlin.util

/**
 * @author Montaser Sbaih
 * @version 1.0
 * @email montaser.jjs@gmail.com
 * @phone +962-786258874
 * @since 20-10-2022
 */

data class TestCase(val input: String, val output: String) {

    override fun toString(): String = "$input\n-\n$output"

    companion object {

        fun parse(testCase: String): TestCase {
            val io = testCase.split("\n*\\-+\n*".toRegex())
            return TestCase(io[0], io[1])
        }
    }
}